### ISSUE TITLE
Argument 1 passed to NunoMaduro\PhpInsights\Domain\Insights\Insight::shouldSkipFile() must be of the type string, int given

### ISSUE THREAD

**Issue by celcius-jochen** [nunomaduro/phpinsights#325]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| Library version  | 1.11.0
| PHP version  | 7.3.5
| OS version  | Microsoft Windows 10 Home (10.0.17763)

Similar to #315 and #318 I'm still getting the following error (tested with custom and default config):
```
  Symfony\Component\Debug\Exception\FatalThrowableError  : Argument 1 passed to NunoMaduro\PhpInsights\Domain\Insights\Insight::shouldSkipFile() must be of the type string, int given, called in ...\vendor\nunomaduro\phpinsights\src\Domain\Insights\Insight.php on line 73

  at ...\vendor\nunomaduro\phpinsights\src\Domain\Insights\Insight.php:55
    51|     {
    52|         return static::class;
    53|     }
    54|
  > 55|     protected function shouldSkipFile(string $file): bool
    56|     {
    57|         $filepath = $file;
    58|         if (mb_strpos($this->collector->getDir(), $file) === false) {
    59|             $filepath = $this->collector->getDir() . DIRECTORY_SEPARATOR . $file;

  Exception trace:

  1   NunoMaduro\PhpInsights\Domain\Insights\Insight::shouldSkipFile()
      ...\vendor\nunomaduro\phpinsights\src\Domain\Insights\Insight.php:73

  2   NunoMaduro\PhpInsights\Domain\Insights\Insight::NunoMaduro\PhpInsights\Domain\Insights\{closure}()
      [internal]:0
```

**Comment by nunomaduro** [comment]
@Jibbarth Could you double check this buddy? 🙏

**Comment by Jibbarth** [comment]
Thank you for reporting this issue @celcius-jochen 

Could you provide your config file and how you launch the analyse ?

**Comment by celcius-jochen** [comment]
I did some debugging and here's what I found so far:

The function `filterFilesWithoutExcluded()` is receiving an array of files, in my case:

```
Array
(
    [0] => ...\app\Enums\ImageType.php
    [1] => ...\app\Enums\VatPercentage.php
)
```

Then `array_filter()` is called with the flag `ARRAY_FILTER_USE_KEY`.

```
return array_filter($files, function ($file): bool {
    return $this->shouldSkipFile($file) === false;
}, ARRAY_FILTER_USE_KEY);
```

This results in `shouldSkipFile()` receiving 0 and 1 (the array indices) instead of the value.
Removing the flag solves the error.

```
return array_filter($files, function ($file): bool {
    return $this->shouldSkipFile($file) === false;
});
```
Then I noticed the following comment:
`@param array<string, string|int|float|array> $files File path must be in key`
So that would probably break other stuff...

The two files that are automatically being excluded are final classes, for example:
```
<?php

namespace App\Enums;

use BenSampo\Enum\Enum;

final class VatPercentage extends Enum
{
    public const VAT_6  = 1.06;
    public const VAT_21 = 1.21;
}
```
Any advice on where to look next?

**Comment by Jibbarth** [comment]
Okay I see. You use the laravel preset, that exclude final classes. 

The bug is so located in [ForbiddenFinalClasses.php#L27-L29](https://github.com/nunomaduro/phpinsights/blob/master/src/Domain/Insights/ForbiddenFinalClasses.php#L27-L29=).

@celcius-jochen Do you want to provide a fix for this ? You can see #316 for inspiration. 

If you cannot, no problem, I'll provide a fix tonight :wink:

**Comment by celcius-jochen** [comment]
That's more than inspiration, I call it copy/paste 😅
Thanks for the quick response and help!

**Comment by Jibbarth** [comment]
Thank you for reporting this issue and also fixing it @celcius-jochen :heart: 

It's now fixed in v1.11.1 :muscle:
